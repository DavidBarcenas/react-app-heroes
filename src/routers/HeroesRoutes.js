import React from 'react'
import { Navbar } from '../components/ui/Navbar'
import { Switch, Route, Redirect } from 'react-router-dom'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { DcScreen } from '../components/dc/DcScreen'

export const HeroesRoutes = () => {
  return (
    <>
      <Navbar />

      <div>
        <Switch>
          <Route exact path="/marvel" component={ MarvelScreen } />
          <Route exact path="/dc" component={ DcScreen } />
          <Route exact path="/heroe/:id" component={ HeroScreen } />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  )
}
