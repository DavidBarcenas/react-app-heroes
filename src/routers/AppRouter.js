import React from 'react'
import { BrowserRouter as Router, Switch } from 'react-router-dom'
import { LoginScreen } from '../components/login/LoginScreen'
import { HeroesRoutes } from './HeroesRoutes'
import { PrivateRoute } from './PrivateRoute'
import { PublicRoute } from './PublicRoute'
import { useContext } from 'react'
import { AuthContext } from '../auth/AuthContext'

export const AppRouter = () => {

  const {user: {logged}} = useContext(AuthContext);

  return (
    <Router>
      <Switch>
        <PublicRoute exact path='/login' isAuth={logged} component={LoginScreen}/>
        <PrivateRoute path='/' isAuth={logged} component={HeroesRoutes}/>
      </Switch>
    </Router>
  )
}
