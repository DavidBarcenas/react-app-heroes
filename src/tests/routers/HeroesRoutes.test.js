import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { HeroesRoutes } from '../../routers/HeroesRoutes';
import { AuthContext } from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';


describe('<HeroesRoutes /> testing', () => {

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  }
  
  test('should render correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <AuthContext.Provider value={contextValue}>
          <HeroesRoutes />
        </AuthContext.Provider>
      </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
  })
  

})
