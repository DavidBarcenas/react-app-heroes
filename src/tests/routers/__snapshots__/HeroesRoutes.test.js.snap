// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroesRoutes /> testing should render correctly 1`] = `
Array [
  <nav
    className="main__nav red"
  >
    <div
      className="nav-wrapper"
    >
      <a
        aria-current={null}
        className="brand-logo"
        href="/"
        onClick={[Function]}
      >
        Heroes
      </a>
      <ul
        className="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="activeUser"
        />
        <li>
          <button
            className="waves-effect waves-light btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="main__views"
  >
    <div>
      <div
        className="heroes__wrapper"
      >
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-spider"
            onClick={[Function]}
          >
            <img
              alt="Spider Man"
              src="./assets/heroes/marvel-spider.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-captain"
            onClick={[Function]}
          >
            <img
              alt="Captain America"
              src="./assets/heroes/marvel-captain.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-iron"
            onClick={[Function]}
          >
            <img
              alt="Iron Man"
              src="./assets/heroes/marvel-iron.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-thor"
            onClick={[Function]}
          >
            <img
              alt="Thor"
              src="./assets/heroes/marvel-thor.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-hulk"
            onClick={[Function]}
          >
            <img
              alt="Hulk"
              src="./assets/heroes/marvel-hulk.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-wolverine"
            onClick={[Function]}
          >
            <img
              alt="Wolverine"
              src="./assets/heroes/marvel-wolverine.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-daredevil"
            onClick={[Function]}
          >
            <img
              alt="Daredevil"
              src="./assets/heroes/marvel-daredevil.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-hawkeye"
            onClick={[Function]}
          >
            <img
              alt="Hawkeye"
              src="./assets/heroes/marvel-hawkeye.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-cyclops"
            onClick={[Function]}
          >
            <img
              alt="Cyclops"
              src="./assets/heroes/marvel-cyclops.jpg"
            />
          </a>
        </div>
        <div
          className="herocard animate__animated animate__fadeIn"
        >
          <a
            href="/hero/marvel-silver"
            onClick={[Function]}
          >
            <img
              alt="Silver Surfer"
              src="./assets/heroes/marvel-silver.jpg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
